actors/src/main/kotlin/ktx/actors/actions.kt

```diff
@@ -33,65 +33,77 @@ operator fun Actor.plusAssign(action: Action) = addAction(action)
 operator fun Actor.minusAssign(action: Action) = removeAction(action)
 
 /**
- * Wraps this action and the passed action with a [SequenceAction].
- *
- * If [this] or [action] are [SequenceAction]s, their nested actions will be unwrapped.
+ * Wraps this action and the passed action in a [SequenceAction].
  *
  * @param action will be executed after this action.
  * @return [SequenceAction] storing both actions.
  */
-infix fun Action.then(action: Action): SequenceAction {
-  val result = Actions.sequence()
-  result.addUnwrapped(this)
-  result.addUnwrapped(action)
-  return result
-}
+infix fun Action.then(action: Action): SequenceAction = Actions.sequence(this, action)
 
-private fun SequenceAction.addUnwrapped(action: Action) {
-  when (action) {
-    is SequenceAction -> action.actions.forEach(::addAction)
-    else -> addAction(action)
-  }
+/**
+ * Adds [action] to this [SequenceAction].
+ *
+ * @param action will be added to this [SequenceAction]
+ * @return [SequenceAction] this [SequenceAction]
+ */
+infix fun SequenceAction.then(action: Action): SequenceAction {
+  addAction(action)
+  return this
 }
 
 /**
- * Wraps this action and the passed action with a [SequenceAction].
- * @param action will be executed after this action.
+ * Wraps this action and the passed action in a [SequenceAction].
+ *
+ * @param action will be executed after [this] action.
  * @return [SequenceAction] storing both actions.
  */
 operator fun Action.plus(action: Action): SequenceAction = then(action)
```

Comment:
Since `then` is now documented as mutating, let's not reuse it in a non-mutating operator and just reimplement it with `Actions.sequence` just to be future-proof.